packages/bolt-connection/test/lang/functional.test.js

```diff
@@ -0,0 +1,170 @@
+import { reuseOngoingRequest } from '../../src/lang/functional.js'
+
+describe('functional', () => {
+  describe('reuseOnGoingRequest', () => {
+    it('should call supplied function with the params', async () => {
+      const expectedParams = ['a', 1, { a: 'a' }]
+      const func = jest.fn(() => Promise.resolve())
+
+      const decoratedFunction = reuseOngoingRequest(func)
+      await decoratedFunction(...expectedParams)
+
+      expect(func).toHaveBeenCalledWith(...expectedParams)
+    })
+
+    it('should call supplied function with this', async () => {
+      const expectedParams = ['a', 1, { a: 'a' }]
+      const thisArg = { t: 'his' }
+      const func = jest.fn(async function () {
+        return thisArg
```

Comment:
```suggestion
        return this
```
or am I misunderstanding things?

Reply:
I didn't misunderstand, the test is wrong.